share/change.py

```diff
@@ -193,7 +218,12 @@ def __init__(self, graph):
         self.__graph = graph
         self.__visted = set()
         self.__visiting = set()
-        self.__nodes = list(graph.nodes)
+
+        # visit identifiers first
+        self.__nodes = sorted(
```

Comment:
Ohh, I missed that the result of resolving those relations is used by the sort (though now it seems obvious). Yeah, the existing logic should be fine.